src/components/DirectorCardsSearch/SearchPlugin.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import directorsStyles from "./directors.module.scss"
+
+class SearchPlugin extends React.Component {
+    constructor(props) {
+      super(props);
+  
+      this.onTextChanged = this.onTextChangedHandler.bind(this);
+      this.search = this.searchHandler.bind(this);
+    }
+  
+    onTextChangedHandler(e) {
+      let searchString = e.target.value.trim();
+      this.search(searchString);
+    }
+  
+    searchHandler(searchString) {
+      let filteredList = this.props.items.filter(item => {
+        return item.node.name.toLowerCase().search(searchString.toLowerCase()) !== -1;
```

Comment:
Change function for search name or city. If u want, add the country by analogy with the city. 
```
let filteredList = this.props.items.filter(item => {
        let directorNameCity = item.node.name.toLowerCase() + item.node.birthCity[0].city.toLowerCase()
        return directorNameCity.search(searchString.toLowerCase()) !== -1;
      });
```